catalog/spatial/wfs/1.1.0/spatial-wfs-v1_1_0-source/src/test/java/org/codice/ddf/spatial/ogc/wfs/v1_1_0/catalog/source/WfsFilterDelegateTest.java

```diff
@@ -0,0 +1,2245 @@
+    List<String> mockTextProps = new ArrayList<>();
+    mockTextProps.add(MOCK_PROPERTY);
+    when(featureMetacardType.getProperties()).thenReturn(mockTextProps);
+    when(featureMetacardType.getAttributeDescriptor(MOCK_PROPERTY))
+        .thenReturn(
+            new FeatureAttributeDescriptor(
+                MOCK_PROPERTY, MOCK_PROPERTY, true, true, true, true, BasicTypes.INTEGER_TYPE));
+    WfsFilterDelegate delegate =
+        new WfsFilterDelegate(featureMetacardType, SUPPORTED_GEO, SRS_NAME);
+    short literal = 1;
+
+    FilterType filter = delegate.propertyIsGreaterThan(MOCK_PROPERTY, literal);
+
+    Writer writer = new StringWriter();
+    Marshaller marshaller = JAXB_CONTEXT.createMarshaller();
+    marshaller.marshal(getFilterTypeJaxbElement(filter), writer);
+    String xml = writer.toString();
+
+    assertXMLEqual(propertyGreaterThanXml, xml);
+  }
+
+  @Test
+  public void testPropertyIsGreaterThanLong() throws JAXBException, SAXException, IOException {
+    List<String> mockTextProps = new ArrayList<>();
+    mockTextProps.add(MOCK_PROPERTY);
+    when(featureMetacardType.getProperties()).thenReturn(mockTextProps);
+    when(featureMetacardType.getAttributeDescriptor(MOCK_PROPERTY))
+        .thenReturn(
+            new FeatureAttributeDescriptor(
+                MOCK_PROPERTY, MOCK_PROPERTY, true, true, true, true, BasicTypes.LONG_TYPE));
+    WfsFilterDelegate delegate =
+        new WfsFilterDelegate(featureMetacardType, SUPPORTED_GEO, SRS_NAME);
+    long literal = 1L;
+    FilterType filter = delegate.propertyIsGreaterThan(MOCK_PROPERTY, literal);
+    Writer writer = new StringWriter();
+    Marshaller marshaller = JAXB_CONTEXT.createMarshaller();
+    marshaller.marshal(getFilterTypeJaxbElement(filter), writer);
+
+    assertXMLEqual(propertyGreaterThanXml, writer.toString());
+  }
+
+  @Test
+  public void testPropertyIsGreaterThanFloat() throws JAXBException, SAXException, IOException {
+    List<String> mockTextProps = new ArrayList<>();
+    mockTextProps.add(MOCK_PROPERTY);
+    when(featureMetacardType.getProperties()).thenReturn(mockTextProps);
+    when(featureMetacardType.getAttributeDescriptor(MOCK_PROPERTY))
+        .thenReturn(
+            new FeatureAttributeDescriptor(
+                MOCK_PROPERTY, MOCK_PROPERTY, true, true, true, true, BasicTypes.LONG_TYPE));
+    WfsFilterDelegate delegate =
+        new WfsFilterDelegate(featureMetacardType, SUPPORTED_GEO, SRS_NAME);
+    float literal = 1.0F;
+    FilterType filter = delegate.propertyIsGreaterThan(MOCK_PROPERTY, literal);
+    Writer writer = new StringWriter();
+    Marshaller marshaller = JAXB_CONTEXT.createMarshaller();
+    marshaller.marshal(getFilterTypeJaxbElement(filter), writer);
+
+    assertXMLEqual(propertyGreaterThanXmlDecimal, writer.toString());
+  }
+
+  @Test
+  public void testPropertyIsGreaterThanDouble() throws JAXBException, SAXException, IOException {
+    List<String> mockTextProps = new ArrayList<>();
```

Comment:
might want to consider moving some of this mock setup in `@Before` or a separate method

Reply:
I worked on it and reduce the line count by 580 lines. Take a look and let me know if you want more refactoring.

Reply:
that should be good